src/phpDocumentor/Configuration/Definition/Version2.php

```diff
@@ -152,7 +152,12 @@ public function getConfigTreeBuilder() : TreeBuilder
      *
      * @param array<string, string|array<string, mixed>> $values
      *
-     * @return array<string, array<int|string, mixed>|string>
+     * @return array{
+     * configVersion: string, title: array<string, mixed>|string, paths: array{output: mixed, cache: mixed},
+     * version: array{0: array{number: string, api: array{0: array{default-package-name: mixed,
```

Comment:
`default-package-name` should be `string|null` 

I do not think we should define `0` as a key here, it is just an array with `int` index type. Is there a specific reason to define it this way? Maybe how we are using it right now in the application?

Reply:
This was Psalm inferring what it can based on the code. I did not look too deep in the array because my goal was to have the first few levels right. I'll revert this part anyway, I already rewrote that in another branch